pageserver/src/walreceiver/walreceiver_connection.rs

```diff
@@ -84,6 +116,14 @@ pub async fn handle_walreceiver_connection(
 
     let identify = identify_system(&mut replication_client).await?;
     info!("{identify:?}");
+
+    connection_status.latest_connection_update = Utc::now().naive_utc();
```

Comment:
Looks like the creation and updates of the `WalConnectionStatus` could be done via the [`send`](https://docs.rs/tokio/latest/tokio/sync/watch/struct.Sender.html#method.send) and [`send_modify`](https://docs.rs/tokio/latest/tokio/sync/watch/struct.Sender.html#method.send_modify`) correspondingly with less code.

Reply:
I tried `send_modify` now, but it requires matching `TaskEvent::NewEvent<E>` before modifying, not very convenient. I think I'll leave it as it is now, if you know how to make it better you can send a PR later.